nautobot/core/views/utils.py

```diff
@@ -274,3 +274,33 @@ def prepare_cloned_fields(instance):
     param_string = urllib.parse.urlencode(params)
 
     return param_string
+
+
+def view_changes_not_saved(request, view, current_saved_view):
+    """
+    Compare request.GET's query dict with the configuration stored on the current saved view
+    If there is any configuration different, return True
+    If every configuration is the same, return False
+    """
+    if current_saved_view is None:
+        return False
+    query_dict = request.GET.dict()
+
+    if query_dict.get("table_changes_pending", None):
+        return True
+    if int(query_dict.get("per_page")) != current_saved_view.config.get("pagination_count", 50):
+        return True
+    if (request.GET.getlist("sort", [])) != current_saved_view.config.get("sort_order", []):
+        return True
+    query_dict_keys = sorted(list(query_dict.keys()))
+    for param in view.non_filter_params:
+        if param in query_dict_keys:
+            query_dict_keys.remove(param)
+    filter_params = current_saved_view.config.get("filter_params", {})
+
+    if set(query_dict_keys) != set(filter_params.keys()):
+        return True
```

Comment:
Similar kind of concern here - if there are no query_dict_keys at all that should be OK right?